CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(sail_riscv)
+
+# Enable CTest
+enable_testing()
+
+# We technically require C++20 since the C generated by Sail - which we compile
+# as C++ - uses designated initialisers, a C++20 feature. However in practice
+# much older compilers support this feature so everything does work with C++17,
+# but you get lots of warnings on compilers that support C++20 if you use
+# this feature without -std=c++20.
+if (cxx_std_20 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+    set(CMAKE_CXX_STANDARD 20)
+else()
+    set(CMAKE_CXX_STANDARD 17)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+# Export compile_commands.json for IDE support.
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+
+# Always use Position Independent Code. By default it is only used for
+# shared libraries (which require it), but you also need it for static
+# libraries if you link them into shared libraries.
+# Generally it just simplifies everything for a negligable performance cost.
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+# Don't allow undefined symbols in shared libraries. This is generally a pain.
+if (UNIX)
+    set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined")
+endif()
+
+# Optional faster binary. Increases compilation time a lot though due to LTO.
+option(EXTRA_FAST "Enable aggressive optimisation flags (-march=native, -flto, etc)" FALSE)
+
+if (EXTRA_FAST)
+    include("cmake/extra_fast.cmake")
+endif()
```

Comment:
I guess one problem with this is that by default CMake will give you a debug build so without instructions we might end up with people complaining that their LTO build is still super slow.

Might need to check that CMAKE_BUILD_TYPE is not empty and not `Debug`?

Reply:
Good point. I added a warning.